pulp_smash/tests/rpm/api_v2/test_unavailable_checksum.py

```diff
@@ -146,3 +173,46 @@ def verify_primary_xml(self, cfg, distributor, checksum_type):
             package.find(xpath).get('type') for package in packages
         }
         self.assertEqual(checksum_types, {checksum_type})
+
+    def verify_filelists_xml(self, cfg, distributor, checksum_type):
+        """Verify a published repo's filelists.xml uses the given checksum."""
+        filelist_xml = get_repodata(cfg, distributor, 'filelists')
+        xpath = '{{{}}}package'.format(RPM_NAMESPACES['metadata/filelists'])
+        packages = filelist_xml.findall(xpath)
+        pkgids_len = {len(package.get('pkgid')) for package in packages}
+        self.verify_pkgid_len(
+            pkgids_len, checksum_type, 'verify_filelists.xml')
+
+    def verify_other_xml(self, cfg, distributor, checksum_type):
+        """Verify a published repo's other.xml uses the given checksum."""
+        other_xml = get_repodata(cfg, distributor, 'other')
+        xpath = '{{{}}}package'.format(RPM_NAMESPACES['metadata/other'])
+        packages = other_xml.findall(xpath)
+        pkgids_len = {len(package.get('pkgid')) for package in packages}
+        self.verify_pkgid_len(pkgids_len, checksum_type, 'verify_other.xml')
+
+    def verify_pkgid_len(self, pkgid_len, checksum_type, comment):
+        """Perform assertion based on given the checksum type.
+
+        Based on the checksum type provided different assertions will
+        be performed over the length of pkgid.
+        """
+        if checksum_type == 'sha256':
+            # sha256 is 256 bit long, and uses 4 bits
+            # then 256/4 = 64
```

Comment:
FYI, this relationship can be expressed algebraicallyl. If a single term (like "bits") is in both a numerator and denominator, it gets crossed out, just like for numbers.

    # 256 bits / (4 bits / 1 hex digit) == 64 hex digits.

What I find nice about a representation like this is that every unit is identified. No need to wonder what "4" corresponds to, or what "64" corresponds to.

Reply:
Updated per comments.